### ISSUE TITLE
accelerate model serving

### ISSUE THREAD

**Issue by WangYongzhao** [OpenNMT/OpenNMT-tf#473]
I was trying to serve a model, currently met latency issue.

So I plan to use "knowledge distilation", refer the paper. Distilling the Knowledge in a Neural Network https://arxiv.org/abs/1503.02531

Anyone tried "knowledge distilation" in opennmt-tf before, or any other suggestion I can accelerate my model?

**Comment by guillaumekln** [comment]
Are you serving the model on CPU or GPU?

**Comment by WangYongzhao** [comment]
hosted on GPU currently.

**Comment by guillaumekln** [comment]
Can you give more information about the model you are serving and its configuration?

**Comment by WangYongzhao** [comment]
my model is similar with https://github.com/OpenNMT/OpenNMT-tf/blob/master/config/models/multi_source_transformer.py , only change share_encoders=False

**Comment by guillaumekln** [comment]
This model is quite compute heavy. I don't think there are magical tricks. You can try reducing the model size, disabling beam search, or... buying more powerful hardware.

**Comment by WangYongzhao** [comment]
thanks, so i was trying to use the idea "knowledge distilation"/"teacher-student", refer the paper. Distilling the Knowledge in a Neural Network https://arxiv.org/abs/1503.02531
so want to know if there are similar works before. or some quick way to test this idea.

**Comment by guillaumekln** [comment]
Yes it works and it is way to reduce the model size without loosing much in accuracy. Also read https://arxiv.org/abs/1606.07947

**Comment by WangYongzhao** [comment]
to try the distilling idea, I want to follow this sample to experiment, but currently met some issues, need debugging.
https://github.com/OpenNMT/OpenNMT-tf/tree/master/examples/library.

do you have other better idea to try this? the model is better compatiable with opennmt-tf, so in this way I can use standard tool to infer/score. http://opennmt.net/OpenNMT-tf/inference.html

**Comment by guillaumekln** [comment]
If you follow "Sequence-Level Knowledge Distillation" from https://arxiv.org/abs/1606.07947, you don't need to run custom code. It is a simple training procedure: "In sequence-level knowledge distillation (center) the student network is trained on the output from beam search of the teacher network that had the highest score".

**Comment by WangYongzhao** [comment]
Yes, that one is straightforward. 
What i mean is word-kd on the left. Just want to know if it's possiable to test this idea easily.

**Comment by guillaumekln** [comment]
It performs worse than the Seq-KD so maybe there is no need to spend time on that.

**Comment by WangYongzhao** [comment]
I see, thanks.
